### ISSUE TITLE
Cocoapods / Podspec for iOS / OS X targets

### ISSUE THREAD

**Issue by chrisballinger** [EricssonResearch/openwebrtc#59]
To ease integration for iOS / OS X targets, a semi-official podspec would be useful for other developers. If I create one, would this be something you would consider merging?

**Comment by superdump** [comment]
Maybe it's nicer to do this once we have binary package releases? Perhaps the framework itself can be hosted somewhere so that a cocoapod could just download that?

**Comment by stefanalund** [comment]
I also think that the pod should include helper classes rather than the OpenWebRTC.framework (which may be quite large). From what I understand pods typically only contain source files. 

Anyway, I'm working on this right now.

**Comment by superdump** [comment]
Ah, so you would have the framework separately and then helper classes in the cocoapod?

**Comment by stefanalund** [comment]
Yes, that is my current thinking. But it would be nice to also somehow automatically get OpenWebRTC.framework installed through CocoaPods. Not sure how to do that yet.

**Comment by stefanalund** [comment]
It may be possible to include the .framework after all: https://github.com/rivera-ernesto/CocoaPods-Specs/blob/master/NewRelicAgent/1.328.0/NewRelicAgent.podspec

**Comment by superdump** [comment]
That's what I was thinking - publish a tarball or zip of the framework itself if possible. Then download and extract it as part of the pod. Though it's not really that difficult to download and install it from a dmg/pkg installer...

**Comment by stefanalund** [comment]
Finally we have some news to share on this topic: https://github.com/EricssonResearch/openwebrtc-ios-sdk

The SDK consists of 2 separate pods, one that contains the pre-built OpenWebRTC.framework based on the output of [Cerbero](https://github.com/EricssonResearch/openwebrtc/wiki/Building-OpenWebRTC-with-Cerbero), and another one with helper classes that makes building apps much simpler. 

I'd say these are 90% done. As you can see in the usage instructions there are still some manual fixes needed in the resulting `.xcworkspace` file needed after each `pod install`. If someone can shed some light on this I would be extremely grateful.

The `OpenWebRTC-SDK` pod is still not published to trunk, but can be used as a development pod.

As these are my first CocoaPod's I would be super happy about any feedback. A sample application using both these pods is available here: https://github.com/EricssonResearch/openwebrtc-examples/pull/49

Keeping this issue open until both pods are in trunk.

Thanks.

**Comment by appden** [comment]
Would you please add a pre-built OS X framework as well. I see that this was the intent in the title of the issue, but as right now, only an iOS framework is available on CocoaPods. I did actually go through the rather painful process of building the framework from scratch, but it would be great to easily upgrade to official releases.

Also, I'm curious if the published iOS framework is a debug build since the download is 135mb (415mb unzipped!). My build of the OS X framework ended up being about 36mb. Thanks!

**Comment by superdump** [comment]
We'll be making a prerelease any day now and then we'll update cocoapods.

The iOS framework has to support multiple architectures - armv7, armv7s, arm64, x86_64 I think. That is why it is so big.

**Comment by appden** [comment]
Thanks for the quick response! I'm actually curious about the size of the framework (even at 36mb) because according to iTunes, Bowser is about 10mb. Does that mean it's possible to further reduce the size of the framework?

**Comment by stefanalund** [comment]
@appden Bowser on the App Store does not yet use the new .framework file. But our [NativeDemo](https://github.com/EricssonResearch/openwebrtc-examples/tree/master/ios/NativeDemo) app does and it is ~20 MB when archiving. I think non-used symbols, documentation and general debug stuff is remved once you build for Release.

**Comment by appden** [comment]
@stefanalund that's great to hear! I look forward to integrating the official release of the framework. :+1

**Comment by stefanalund** [comment]
@appden you may have misunderstood me :) What I meant is that even though the framework is quite huge, the compiled application is not.

**Comment by appden** [comment]
@stefanalund I don't think I misunderstood. I'm happy to hear that Bowser compiles down to around 20mb. My original question was about using an official release of the OS X framework. That's what I was referencing. :)

**Comment by shoreflyer** [comment]
Hi, I've had to compile openwebrtc myself due to dependancies on owr-gst. How do I best go about using cocoa pods to integrate the built and now locally installed frameworks?

**Comment by shoreflyer** [comment]
Okay, I think I have answered my own question and would like to share my findings.
In the developer podspec change 
```
s.source = { :http => "http://static.verkstad.net/OpenWebRTC_0.3.1.zip" }
```
to
```
s.source = { :file => "file:~/Library/Developer/OpenWebRTC/iPhone.sdk" }
```
if you have installed the framework like documented on the building openwebrtc.
I've basically tried my luck with the :file as it wasn't documented anywhere I could tell in the cocoa pods documentation.

**Comment by superdump** [comment]
@shoreflyer - it may be easier to not use cocoapods in your case and add the frameworks and system libraries/frameworks manually.
